### PR TITLE
Distinguish between internal and user `repair_arg` errors

### PR DESCRIPTION
**Pull request by lionel-**
Follow up to #1662

### CODE REVIEW

File: src/names.c [hunk 1]
```diff
@@ -820,13 +866,13 @@ struct name_repair_opts new_name_repair_opts(r_obj* name_repair,
     .fn = r_null,
     .name_repair_arg = name_repair_arg,
     .quiet = quiet,
-    .call = call
+    .call = call,
```
Comment (DavisVaughan) [inline src/names.c:869]: Purposeful?

Comment (DavisVaughan) [reply src/names.c:869]: I don't see the new `.frame` struct element being used anywhere?

**Review by DavisVaughan — APPROVED** [review]

File: src/names.c (hunk 1 above)
Comment (lionel-) [reply src/names.c:869]: It is used in `new_repair_error_info()` to create an internal error call.

Comment (DavisVaughan) [reply src/names.c:869]: Yes, but I don't see where you ever set the `frame` element when you create a new `struct name_repair_opts`

Comment (lionel-) [reply src/names.c:869]: oh yeah what happened to that arg. Hmm I'll just call `peek_frame()`

Comment (lionel-) [reply src/names.c:869]: but what if `vec_as_names()` was called from C :-(
This stuff is too hard.

File: src/names.c [hunk 2]
```diff
@@ -22,30 +22,82 @@ r_obj* vec_as_names(r_obj* names, const struct name_repair_opts* opts) {
...
+
+    // This won't be set when `vec_as_names()` is called from C
+    if (p_opts->frame) {
+      out.input_error_call = p_opts->frame;
+      r_list_poke(out.shelter, 3, out.input_error_call);
+    } else {
+      // Create fake `vec_as_names()` call for the C case
+      out.input_error_call = r_call(r_sym("vec_as_names"));
+    }
+    r_list_poke(out.shelter, 3, out.input_error_call);
```
Comment (DavisVaughan) [inline src/names.c:87]: Looks like you are double poking

**Review by DavisVaughan — APPROVED** [review]
LGTM now besides the double poke i mentioned

**Pull request merged**